dask_distance/__init__.py

```diff
@@ -160,12 +160,9 @@ def pdist(X, metric="euclidean", **kwargs):
 
     result = cdist(X, X, metric, **kwargs)
 
-    result = dask.array.triu(result, 1)
-
-    indices = _compat._indices(result.shape, chunks=result.chunks)
-    mask = (indices[1] > indices[0])
-
-    result = _compat._ravel(result)[_compat._ravel(mask)]
+    result = dask.array.concatenate([
+        result[i, i + 1:] for i in range(0, len(result) - 1)
+    ])
```

Comment:
Am a little concerned about this performance-wise for large numbers of points. Reason being this makes the graph balloon with `getitem` entries. Would be good if we could cut this down somehow, but it is not obvious to me how without reusing the old masking strategy.

Reply:
That said, it seems to do ok given reasonable chunk sizes when playing around with it locally. So perhaps this is not worth worrying about until use cases that have issues present themselves.